certora/WormholeOracleAuth.spec

```diff
@@ -0,0 +1,301 @@
+}
+
+// Verify that removeSigners behaves correctly
+rule removeSigners(address[] signers_) {
+    env e;
+
+    uint256 i;
+    uint256 length = signers_.length;
+
+    require(i < length);
+
+    removeSigners(e, signers_);
+
+    assert(signers(signers_[i]) == 0, "addSigners did not set signer as expected");
+}
+
+// Verify revert rules on removeSigners
+rule removeSigners_revert(address[] signers_) {
+    env e;
+
+    uint256 ward = wards(e.msg.sender);
+
+    aux.checkMalformedArray@withrevert(signers_);
+    bool malformed = lastReverted; // Nasty workaround to catch malformed array revert case
+
+    removeSigners@withrevert(e, signers_);
+
+    bool revert1 = e.msg.value > 0;
+    bool revert2 = ward != 1;
+    bool revert3 = malformed;
+
+    assert(revert1 => lastReverted, "revert1 failed");
+    assert(revert2 => lastReverted, "revert2 failed");
+    assert(revert3 => lastReverted, "revert3 failed");
+
+    assert(lastReverted => revert1 || revert2 || revert3, "Revert rules are not covering all the cases");
+}
+
+// Verify that requestMint behaves correctly
+rule requestMint(
+        oracle.WormholeGUID guid,
+        bytes signatures,
+        uint256 maxFeePercentage,
+        uint256 operatorFee
+    ) {
+    env e;
+
+    require(wormholeJoin() == join);
+
+    uint256 postFeeAmount;
+    uint256 totalFee;
+    postFeeAmount, totalFee = requestMint(e, guid, signatures, maxFeePercentage, operatorFee);
+
+    bytes32 sourceDomain;
+    bytes32 targetDomain;
+    bytes32 receiver;
+    bytes32 operator;
+    uint128 amount;
+    uint80 nonce;
+    uint48 timestamp;
+    sourceDomain, targetDomain, receiver, operator, amount, nonce, timestamp = join.wormholeGUID();
+    assert(sourceDomain == guid.sourceDomain, "guid.sourceDomain was not preserved");
+    assert(targetDomain == guid.targetDomain, "guid.targetDomain was not preserved");
+    assert(receiver == guid.receiver, "guid.receiver was not preserved");
+    assert(operator == guid.operator, "guid.operator was not preserved");
+    assert(amount == guid.amount, "guid.amount was not preserved");
+    assert(nonce == guid.nonce, "guid.nonce was not preserved");
+    assert(timestamp == guid.timestamp, "guid.timestamp was not preserved");
+    assert(join.maxFeePercentage() == maxFeePercentage, "maxFeePercentage was not preserved");
+    assert(join.operatorFee() == operatorFee, "operatorFee was not preserved");
+    assert(join.postFeeAmount() == postFeeAmount, "postFeeAmount was not preserved");
+    assert(join.totalFee() == totalFee, "totalFee was not preserved");
+}
+
+// Verify revert rules on requestMint
+rule requestMint_revert(
+        oracle.WormholeGUID guid,
+        bytes signatures,
+        uint256 maxFeePercentage,
+        uint256 operatorFee
+) {
+    env e;
+
+    require(guid.amount <= max_uint128);
+    require(guid.nonce <= 0xffffffffffffffffffff);
+    require(guid.timestamp <= 0xffffffffffff);
```

Comment:
These also seem to make no difference now.

Reply:
Removed!